tosan-httpserver-spring-boot-starter/src/main/java/com/tosan/http/server/starter/util/JsonHttpLogContentProvider.java

```diff
@@ -1,28 +1,47 @@
 package com.tosan.http.server.starter.util;
 
+import com.tosan.http.server.starter.wrapper.LogContentContainer;
+
 import java.util.LinkedHashMap;
 import java.util.Map;
 
 /**
  * @author AmirHossein ZamanZade
  * @since 10/29/2022
  */
-public class JsonHttpLogContentProvider extends LogContentProvider<LinkedHashMap<String, Object>> {
-
-    @Override
-    public LinkedHashMap<String, Object> getContentContainer() {
-        return new LinkedHashMap<>();
-    }
+public class JsonHttpLogContentProvider extends LogContentProvider {
 
     @Override
-    public void addToContent(String key, Object value, LinkedHashMap<String, Object> container) {
-        container.put(key, value);
+    protected String generateRequestLogContent(LogContentContainer container) {
+        Map<String, Object> requestContent = new LinkedHashMap<>(1);
+        Map<String, Object> objectMap = new LinkedHashMap<>();
+        objectMap.put("service", container.getUrl());
+        objectMap.putAll(container.getHeaders());
+        if (container.hasError()) {
+            objectMap.putAll(container.getErrorParam());
+        } else {
+            if (container.isFormBody()) {
+                objectMap.put("form parameters", container.getBody());
+            } else {
+                objectMap.put("body", container.getBody());
+            }
+        }
+        requestContent.put("Http Request", objectMap);
+        return ToStringJsonUtil.toJson(requestContent);
     }
 
     @Override
-    public String generateLogContent(String key, LinkedHashMap<String, Object> container) {
-        Map<String, Object> finalContent = new LinkedHashMap<>(1);
-        finalContent.put(key, container);
-        return ToStringJsonUtil.toJson(finalContent);
+    protected String generateResponseLogContent(LogContentContainer container) {
+        Map<String, Object> responseContent = new LinkedHashMap<>(1);
+        Map<String, Object> objectMap = new LinkedHashMap<>();
+        objectMap.put("status", container.getStatus());
+        objectMap.putAll(container.getHeaders());
+        if (container.hasError()) {
+            objectMap.putAll(container.getErrorParam());
+        } else {
+            objectMap.put("body", container.getBody());
+        }
+        responseContent.put("Http Response", objectMap);
+        return ToStringJsonUtil.toJson(responseContent);
```

Comment:
mask is done before reaching to this logProvider so no need to use ToStringJsonUtil that try to mask the fields again.this might break the masking functionality. use your own provided object mapper .